File: backend/src/main/scala/microservices/blog/objects/SaveBlogDraftPlannerRequest.scala
// SaveBlogDraftPlannerRequest：博客域保存博客草稿请求对象。

package com.typesafe.travel.blog.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class SaveBlogDraftPlannerRequest(
    userId: String,
    postId: Option[String],
    title: String,
    summary: String,
    coverText: String,
    content: String,
    images: List[BlogImagePlannerResponse],
    tags: List[BlogTagPlannerResponse],
    travelCity: Option[String],
    travelCities: Option[List[String]]
)
object SaveBlogDraftPlannerRequest:
  given sourceEncoder: Encoder[SaveBlogDraftPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[SaveBlogDraftPlannerRequest] = deriveDecoder

File: frontend/src/microservices/blog/objects/SaveBlogDraftPlannerRequest.ts
// SaveBlogDraftPlannerRequest：博客域保存博客草稿请求对象。

import type { ContentImagePlannerResponse } from '@/microservices/content/objects/ContentImagePlannerResponse'
import type { BlogTagResponse } from './BlogTagResponse'

export type SaveBlogDraftPlannerRequest = {
  userId: string
  postId?: string | null
  title: string
  summary: string
  coverText: string
  content: string
  images: ContentImagePlannerResponse[]
  tags: BlogTagResponse[]
  travelCity?: string | null
  travelCities?: string[] | null
}

export const saveBlogDraftPlannerRequestFromJson = (json: string): SaveBlogDraftPlannerRequest =>
  JSON.parse(json) as SaveBlogDraftPlannerRequest

export const saveBlogDraftPlannerRequestToJson = (value: SaveBlogDraftPlannerRequest): string =>
  JSON.stringify(value)